test/hack_inserts_undef.cl

```diff
@@ -72,7 +72,7 @@ kernel void foo(global S* data, float f) {
 // CHECK-DAG: [[_uint_3:%[0-9a-zA-Z_]+]] = OpConstant [[_uint]] 3
 // CHECK-DAG: [[_float_2:%[0-9a-zA-Z_]+]] = OpConstant [[_float]] 2
 // CHECK-DAG: [[_float_1:%[0-9a-zA-Z_]+]] = OpConstant [[_float]] 1
-// CHECK-DAG: [[_float_0:%[0-9a-zA-Z_]+]] = OpConstant [[_float]] 0
+// CHECK-DAG: [[_undef:%[0-9a-zA-Z_]+]] = OpUndef [[_float]]
```

Comment:
Some drivers will also require -hack-undef, which will change this to the constant 0 anyway.